plugins/module_utils/scaleway.py

```diff
@@ -176,8 +187,24 @@ def ok(self):
 class Scaleway:
     def __init__(self, module: AnsibleModule) -> None:
         self.module = module
+        oauth_token = self.module.params.get("api_token")
+        scw_profile = self.module.params.get("profile")
+
+        if scw_profile:
+            if "SCW_CONFIG_PATH" in os.environ:
+                scw_config_path = os.getenv("SCW_CONFIG_PATH", "")
+            elif "XDG_CONFIG_HOME" in os.environ:
+                scw_config_path = os.path.join(os.getenv("XDG_CONFIG_HOME", ""), "scw", "config.yaml")
+            else:
+                scw_config_path = os.path.join(os.path.expanduser("~"), ".config", "scw", "config.yaml")
+
+            if os.path.exists(scw_config_path):
+                with open(scw_config_path) as fh:
+                    scw_config = yaml.safe_load(fh)
```

Comment:
You don't check `YAML_IMPORT_ERROR` before using `yaml`. `yaml` is undefined if `YAML_IMPORT_ERROR` is not `None`.

Reply:
Is that not fixed by this, right at the top of the script? https://github.com/ansible-collections/community.general/pull/11314/files#diff-4c51452de57fcb64cc7be370d5d79d37868a8e374b1742c358a84a6ec5a720b9R17

If not, I'm not quite sure how to do this, sorry.

Reply:
Check out https://docs.ansible.com/projects/ansible/devel/dev_guide/testing/sanity/import.html#in-modules

I'd do this at the beginning fo the `if scw_profile:`.

You also have to add this as `requirements:` in plugins/doc_fragments/scaleway.py, like:
```
requirements:
  - PyYAML (when O(profile) is used)
```

Reply:
Oh, I see what you mean! OK, I'll fix that now. :slightly_smiling_face: